crates/cxx-qt-gen/src/naming/mod.rs

```diff
@@ -29,24 +29,76 @@ use crate::syntax::{
 
 use crate::parser::{cxxqtdata::ParsedCxxQtData, qenum::ParsedQEnum, qobject::ParsedQObject};
 
+/// This struct contains all names a certain syntax element may have
+///
+/// This includes the rust_name, cxx_name, as well as qualifications like
+/// the rust module and C++ namespace.
+#[derive(Debug)]
+pub struct Name {
```

Comment:
can we split Name into it's own file ? Personally I think we should be heading to almost one struct per file (this is how i see other large projects structured, like CXX tends to do this)

Reply:
Done